lib/jxl/decode_test.cc

```diff
@@ -1540,9 +1543,19 @@ TEST_P(DecodeTestParam, PixelTest) {
         xsize * 2 * orig_channels, nullptr, pixels.data(), pixels.size(),
         nullptr, nullptr, static_cast<jxl::Orientation>(config.orientation)));
   }
-
-  EXPECT_EQ(0u, ComparePixels(pixels.data(), pixels2.data(), xsize, ysize,
-                              format_orig, format));
+  if (config.upsampling == 1) {
+    EXPECT_EQ(0u, ComparePixels(pixels.data(), pixels2.data(), xsize, ysize,
+                                format_orig, format));
+  } else {
+    // resampling is of course not lossless, so as a rough check:
+    // count pixels that are more than off-by-25 in the 8-bit value of one of
+    // the channels
+    EXPECT_LE(
+        ComparePixels(
+            pixels.data(), pixels2.data(), xsize, ysize, format_orig, format,
+            50.0 * (config.data_type == JXL_TYPE_UINT8 ? 1.0 : 256.0)),
+        300u);
```

Comment:
Maybe this amount could be made dependent on the xsize and ysize rather than hardcoded 300? Some % of pixels allowed to differ

Reply:
It also depends on the upsampling factor: iirc about 150 differed with 2x, about 200 with 4x, and about 270 with 8x.

The bounds can be tightened but to find the upsampling bug they are good enough: in the buggy codepath, there are 9000 pixels wrong or so (basically at least 3/4 of the image remains just black...)